internal/pkg/cache/redis/redis.go

```diff
@@ -39,22 +42,42 @@ func NewRedis(address string) *Redis {
 
 // NewRedisWithConfig creates a new Redis instance with configurable options.
 func NewRedisWithConfig(cfg *config.RedisConfig) *Redis {
-	o := &redis.Options{
-		Addr:         cfg.Address,
-		MinIdleConns: cfg.MinIdleConns,
-		PoolSize:     cfg.PoolSize,
-		IdleTimeout:  cfg.IdleTimeout,
-		MaxConnAge:   cfg.MaxConnAge,
+	o := &redis.UniversalOptions{
+		Addrs:           parseRedisAddress(cfg.Address),
+		MinIdleConns:    cfg.MinIdleConns,
+		PoolSize:        cfg.PoolSize,
+		ConnMaxIdleTime: cfg.IdleTimeout,
+		ConnMaxLifetime: cfg.MaxConnAge,
+	}
+
+	c := redis.NewUniversalClient(o)
+
+	err := redisotel.InstrumentMetrics(c)
+	if err != nil {
+		log.Fatalf("got error adding metric instrumentation to redis client: %v", err)
```

Comment:
This is highly unlikely to happen, but would you prefer to fail eagerly (As it is now) or to warn about it and let it go?